### PR TITLE
[pytorch][mobile] deprecate the LLVM-based static analyzer

### PR DESCRIPTION
**Pull request by ljk53**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68180

Since we've open sourced the tracing-based selective build, we can deprecate the
op-dependency-graph-based selective build and the static analyzer tool that
produces the dependency graph.

Differential Revision: [D32358467](https://our.internmc.facebook.com/intern/diff/D32358467/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32358467/)!

### CODE REVIEW

File: .jenkins/pytorch/build-mobile.sh [hunk 1]
```diff
@@ -26,11 +26,6 @@ retry pip install --pre torch torchvision \
 # binary, and running forward pass with a real model.
 if [[ "$BUILD_ENVIRONMENT" == *-mobile-custom-build-static* ]]; then
   TEST_CUSTOM_BUILD_STATIC=1 test/mobile/custom_build/build.sh
```
Comment (iseeyuan) [inline .jenkins/pytorch/build-mobile.sh:28]: Maybe we could change the name later. "TEST_CUSTOM_BUILD_STATIC" looks confusing that we are still doing static selective build? cc @cccclai

Comment (iseeyuan) [reply .jenkins/pytorch/build-mobile.sh:28]: Oh, it may be related to selective build with static dispatch.

**Review by iseeyuan — APPROVED** [review]
LGTM

**Review by seemethere — APPROVED** [review]

File: .jenkins/pytorch/build-mobile.sh (hunk 1 above)
Comment (ljk53) [reply .jenkins/pytorch/build-mobile.sh:28]: Yes, it's selective build with static dispatch. The static analyzer one is below, called "mobile-custom-build-dynamic", as it's created for dynamic dispatch.

**Pull request closed without merging**